app/use_cases/accounts/apps/evolution_apis/message/delivery_job.rb

```diff
@@ -0,0 +1,24 @@
+class Accounts::Apps::EvolutionApis::Message::DeliveryJob < ApplicationJob
+
+  self.queue_adapter = :good_job
+  def perform(event_id)
+    event = Event.find(event_id)
+    if should_delivery?(event)
+      result = Accounts::Apps::EvolutionApis::Message::Export.call(
+        event.app,
+        event.contact.phone,
+        event.content.body.to_plain_text
+      )
+      if result.key?(:ok)
+        event.done = true
```

Comment:
Aqui acho que é interessante gravar o ID da mensagem enviada e quando receberos um webhook sobre essa mensagem, ajustar para não duplicar ela

Reply:
No teste que eu fiz ela n estava sendo duplicada, pois o nome do evento do webhook é `mensages.update` se eu n me engano, e a gnt está tratando somente os webhooks com o evento de `messages.upsert`:  
```ruby
 elsif webhook['event'] == 'messages.upsert'
      if  webhook['data']['messageType'] == 'extendedTextMessage'
        Accounts::Apps::EvolutionApis::Webhooks::Events::ImportMessage.call(evolution_api, webhook,
                                                                            webhook['data']['message']['extendedTextMessage']['text'])
      elsif webhook['data']['messageType'] == 'conversation'
        Accounts::Apps::EvolutionApis::Webhooks::Events::ImportMessage.call(evolution_api, webhook,
                                                                            webhook['data']['message']['conversation'])
      end
    end
```

Reply:
Sim mas é interessante termos o ID da mensagem para poder linkar outros tipos de eventos.
Por exemplo quando uma mensagem for entregue ou lida podemos capturar esse evento e colocar a informação no evento correspondende

Reply:
Tem razão, fiz a modificação nos dois pull requests para adicionar no additional_attributes um campo message_id, nele iremos armazenar o id da mensagem do wpp. Criei um scope chamado by_message_id, onde consultaremos um evento que tenha o id da mensagem, esse scope eu fiz com o intuito de colocar um if consultando se a mensagem existe, se existir ele n passará pelos ifs do do message.upsert. Mas acabei removendo esse if pois como eu falei anteriormente, as mensagens não irão ser duplicadas por conta do tipo do evento, e tbm seria desnecessário toda vez q receber o webhook, fazer essa consulta no banco de dados. Se vc achar interessante colocar ele, é só avisar. O código que eu tinha feito era assim: 
```ruby
 elsif webhook['event'] == 'messages.upsert'
      if evolution_api.account.events.by_message_id(webhook['data']['key']['id']).blank?
        if  webhook['data']['messageType'] == 'extendedTextMessage'
          Accounts::Apps::EvolutionApis::Webhooks::Events::ImportMessage.call(evolution_api, webhook,
                                                                              webhook['data']['message']['extendedTextMessage']['text'])
        elsif webhook['data']['messageType'] == 'conversation'
          Accounts::Apps::EvolutionApis::Webhooks::Events::ImportMessage.call(evolution_api, webhook,
                                                                            webhook['data']['message']['conversation'])
      end
    end
  end
``` 
O scope eu fiz desse jeito: 
```ruby
  scope :by_message_id, -> (message_id) {
    where("additional_attributes ->> 'message_id' = ?", message_id)
  }
```

Reply:
Ficou assim: 
```ruby
 if result.key?(:ok)
  event.done = true
  event.additional_attributes.merge!({ 'message_id' => result[:ok]['key']['id']})
  event.save!
 return { ok: event }
```